js/src/Ice/ConnectionI.js

```diff
@@ -42,10 +42,13 @@ import "./CurrentExtensions.js";
 const StateNotInitialized = 0;
 const StateNotValidated = 1;
 const StateActive = 2;
+// StateHolding is a very transient state in JS: the connection is in StateHolding once the connection establishment
+// has succeeded and until the outgoing connection factory activates the connection - which is essentially immediate.
```

Comment:
The hold -> active logic is still used by the OutgoingConnectionFactory, like in other language mappings.

This occurs only once.